virtual_rainforest/core/config.py

```diff
@@ -100,95 +88,122 @@ def log_all_validation_errors(
 
     Args:
         module_name: The name to register the schema under
+        schema_file_path: The file path to the JSON Schema file for the model
+
     Raises:
-        ValueError: If the schema name has already been used
-        OSError: If the module schema is not valid JSON
-        KeyError: If a module schema is missing one of the required keys
+        ValueError: If the module name has already been used to register a schema
     """
 
-    def wrap(func: Callable) -> Callable:
-        # Type the exception raising with a general base class
-        to_raise: Exception
+    if module_name in SCHEMA_REGISTRY:
+        excep = ValueError(f"The module schema for {module_name} is already registered")
+        LOGGER.critical(excep)
+        raise excep
 
-        if module_name in SCHEMA_REGISTRY:
-            to_raise = ValueError(
-                f"The module schema {module_name} is used multiple times, this "
-                f"shouldn't be the case!",
-            )
-            LOGGER.critical(to_raise)
-            raise to_raise
-        else:
-            # Check that this is a valid schema
-            try:
-                Draft202012Validator.check_schema(func())
-            except exceptions.SchemaError:
-                to_raise = OSError(f"Module schema {module_name} not valid JSON!")
-                LOGGER.critical(to_raise)
-                raise to_raise
+    try:
+        SCHEMA_REGISTRY[module_name] = get_schema(module_name, schema_file_path)
+    except Exception as excep:
+        LOGGER.critical(f"Schema registration for {module_name} failed: check log")
+        raise excep
 
-            # Check that relevant keys are included
-            try:
-                func()["properties"][module_name]
-                func()["required"]
-            except KeyError as err:
-                to_raise = KeyError(
-                    f"Schema for {module_name} module incorrectly structured, {err} key"
-                    f" missing!"
-                )
-                LOGGER.critical(to_raise)
-                raise to_raise
+    LOGGER.info("Schema registered for module %s: %s ", module_name, schema_file_path)
+
+
+def get_schema(module_name: str, schema_file_path: Path) -> dict:
+    """Function to load the JSON schema for a module.
+
+    This function tries to load a JSON schema file and then - if the JSON loaded
+    correctly - checks that the JSON provides a valid JSON Schema.
+
+    Args:
+        module_name: The name to register the schema under
+        schema_file_path: The file path to the JSON Schema file
+
+    Raises:
+        ValueError: the module name has no registered schema path
+        FileNotFoundError: the schema path does not exist
+        JSONDecodeError: the file at the schema path is not valid JSON
+        SchemaError: the file contents are not valid JSON Schema
+        ValueError: the JSON Schema is missing required keys
```

Comment:
Should these two Value errors be grouped, or are they better left separate?

Reply:
Actually, I think one of them is redundant! Good spot.